### PR TITLE
[569058] Special character handling in search index

### PR DESCRIPTION
**Pull request by arnauddieumegard**
- Fix special characters handling in search index search field

Change-Id: I3953d41f61b4f7a1120f77710d9bf716cc1c4f9d
Signed-off-by: Arnaud Dieumegard <arnaud.dieumegard@obeo.fr>

### CODE REVIEW

File: Docgen/business/plugins/org.polarsys.kitalpha.doc.gen.business.core/templates/pattern._0T5DwF4yEeGfnvXFIF-hXQ/method._0T5DxV4yEeGfnvXFIF-hXQ.pt [hunk 1]
```diff
@@ -50,7 +50,7 @@ int i = 0;
 for(Entry<String, IndexItem> entry : indexItems.entrySet()) {
 		i++;
 		%>
-		conceptsList[<%=listCount%>][<%=count%>][0] = "<%=EscapeChars.removeNewline(EscapeChars.forHTML(entry.getValue().getConceptName().replace("\"", "\\\"")))%>";
```
Comment (bzendagui) [inline Docgen/business/plugins/org.polarsys.kitalpha.doc.gen.business.core/templates/pattern._0T5DwF4yEeGfnvXFIF-hXQ/method._0T5DxV4yEeGfnvXFIF-hXQ.pt:53]: What are the impacts of this modification? can all special characters be added in the generated html page?

Comment (arnauddieumegard) [reply Docgen/business/plugins/org.polarsys.kitalpha.doc.gen.business.core/templates/pattern._0T5DwF4yEeGfnvXFIF-hXQ/method._0T5DxV4yEeGfnvXFIF-hXQ.pt:53]: Yes in fact they can.

The modification integrated in commit 57c624f have been tested with the following:

image

Comment (arnauddieumegard) [reply Docgen/business/plugins/org.polarsys.kitalpha.doc.gen.business.core/templates/pattern._0T5DwF4yEeGfnvXFIF-hXQ/method._0T5DxV4yEeGfnvXFIF-hXQ.pt:53]: And it works !

**Review by bzendagui — APPROVED** [review]
Ok for me

**Pull request merged**